assets/src/edit-story/components/panels/elementAlignment.js

```diff
@@ -0,0 +1,333 @@
+          x,
+          y,
+          width,
+          height
+        );
+        offSetX = (newWidth - width) / 2;
+      }
+      return {
+        x: boundRect.startX + offSetX,
+      };
+    });
+  };
+
+  const handleAlignCenter = () => {
+    const centerX = (boundRect.endX + boundRect.startX) / 2;
+    onSetProperties((properties) => {
+      const { width } = properties;
+      return {
+        x: centerX - width / 2,
+      };
+    });
+  };
+
+  const handleAlignRight = () => {
+    onSetProperties((properties) => {
+      const { x, y, width, height, rotationAngle } = properties;
+      let offSetX = 0;
+      if (rotationAngle) {
+        const { width: newWidth } = calcRotatedObjectPositionAndSize(
+          rotationAngle,
+          x,
+          y,
+          width,
+          height
+        );
+        offSetX = (newWidth - width) / 2;
+      }
+      return {
+        x: boundRect.endX - width - offSetX,
+      };
+    });
+  };
+
+  const handleAlignTop = () => {
+    onSetProperties((properties) => {
+      const { x, y, width, height, rotationAngle } = properties;
+      let offSetY = 0;
+      if (rotationAngle) {
+        const { height: newHeight } = calcRotatedObjectPositionAndSize(
+          rotationAngle,
+          x,
+          y,
+          width,
+          height
+        );
+        offSetY = (newHeight - height) / 2;
+      }
+      return {
+        y: boundRect.startY + offSetY,
+      };
+    });
+  };
+
+  const handleAlignMiddle = () => {
+    const centerY = (boundRect.endY + boundRect.startY) / 2;
+    onSetProperties((properties) => {
+      const { height } = properties;
+      return {
+        y: centerY - height / 2,
+      };
+    });
+  };
+
+  const handleAlignBottom = () => {
+    onSetProperties((properties) => {
+      const { x, y, width, height, rotationAngle } = properties;
+      let offSetY = 0;
+      if (rotationAngle) {
+        const { height: newHeight } = calcRotatedObjectPositionAndSize(
+          rotationAngle,
+          x,
+          y,
+          width,
+          height
+        );
+        offSetY = (newHeight - height) / 2;
+      }
+      return {
+        y: boundRect.endY - height - offSetY,
+      };
+    });
+  };
+
+  const handleHorizontalDistribution = () => {
+    const offSetWidth = dataPixels(
+      (boundRect.endX - boundRect.startX) / (selectedElements.length - 1)
+    );
+    onSetProperties((properties) => {
+      const { id, x, y, width, height, rotationAngle } = properties;
+      const elementIndex = selectedElements.findIndex((item) => item.id === id);
+      if (elementIndex === 0 || elementIndex === selectedElements.length - 1) {
+        let offSetX = 0;
+        if (rotationAngle) {
+          const { width: newWidth } = calcRotatedObjectPositionAndSize(
+            rotationAngle,
+            x,
+            y,
+            width,
+            height
+          );
+          offSetX = (newWidth - width) / 2;
+        }
+        return {
```

Comment:
I feel like I'm missing something here, but it'd seem that, when we distribute, the left-most and right-most elements should not move at all?

Reply:
That makes sense for me. Thank you